src/include/common/container/concurrent_bitmap.h

```diff
@@ -204,12 +206,8 @@ class RawConcurrentBitmap {
     T bits = reinterpret_cast<const std::atomic<T> *>(&bits_[*byte_pos])->load();
     if (bits == static_cast<T>(-1)) {
       *byte_pos += static_cast<uint32_t>(sizeof(T));
-      // prevent underflow
-      if (*bits_left < sizeof(T) * BYTE_SIZE) {
-        *bits_left = 0;
-      } else {
-        *bits_left = *bits_left - static_cast<uint32_t>(sizeof(T) * BYTE_SIZE);
-      }
+      // Assumes IsAlignedAndFits<T> is called before, so no need to check for underflow.
+      *bits_left = *bits_left - static_cast<uint32_t>(sizeof(T) * BYTE_SIZE);
```

Comment:
Can you document the assumptions and the purpose of this a little more clearly? It is not immediately obvious to me this is true.